metricsaggregator/src/main/java/io/dockstore/metricsaggregator/client/cli/TerraMetricsSubmitter.java

```diff
@@ -0,0 +1,572 @@
+            .toFormatter();
+    private final MetricsAggregatorConfig config;
+    private final SubmitTerraMetrics submitTerraMetricsCommand;
+    private final AtomicInteger numberOfExecutionsProcessed = new AtomicInteger(0);
+    private final AtomicInteger numberOfExecutionsSubmitted = new AtomicInteger(0);
+    private final AtomicInteger numberOfExecutionsSkipped = new AtomicInteger(0);
+
+    // Keep track of sourceUrls that are found to be ambiguous
+    private final Map<String, String> skippedSourceUrlsToReason = new ConcurrentHashMap<>();
+    // Map of source url to TRS info that was previously calculated
+    private final Map<String, SourceUrlTrsInfo> sourceUrlToSourceUrlTrsInfo = new ConcurrentHashMap<>();
+    // Map of workflow path prefixes, like github.com/organization/repo, to published workflows with the same workflow path prefix
+    private final Map<String, List<MinimalWorkflowInfo>> workflowPathPrefixToWorkflows = new ConcurrentHashMap<>();
+
+    public TerraMetricsSubmitter(MetricsAggregatorConfig config, SubmitTerraMetrics submitTerraMetricsCommand) {
+        this.config = config;
+        this.submitTerraMetricsCommand = submitTerraMetricsCommand;
+    }
+
+    public void submitTerraMetrics() {
+        ApiClient apiClient = setupApiClient(config.getDockstoreServerUrl(), config.getDockstoreToken());
+        ExtendedGa4GhApi extendedGa4GhApi = new ExtendedGa4GhApi(apiClient);
+        WorkflowsApi workflowsApi = new WorkflowsApi(apiClient);
+
+        // Read CSV file
+        Iterable<CSVRecord> workflowMetricRecords;
+        final String inputDateFilePath = this.submitTerraMetricsCommand.getDataFilePath();
+        try (BufferedReader metricsBufferedReader = new BufferedReader(new FileReader(inputDateFilePath))) {
+            final CSVFormat csvFormat = CSVFormat.DEFAULT.builder()
+                    .setHeader(TerraMetricsCsvHeaders.class)
+                    .setSkipHeaderRecord(true)
+                    .setTrim(true)
+                    .build();
+            workflowMetricRecords = csvFormat.parse(metricsBufferedReader);
+
+            // This output file is used to record skipped executions
+            final String outputFileName = inputDateFilePath + "_skipped_executions_" + Instant.now().truncatedTo(ChronoUnit.SECONDS).toString().replace("-", "").replace(":", "") + ".csv";
+
+            List<CSVRecord> workflowMetricsToProcess = new ArrayList<>();
+            try (CSVPrinter skippedExecutionsCsvPrinter = submitTerraMetricsCommand.isRecordSkippedExecutions() ? new CSVPrinter(
+                    new FileWriter(outputFileName, StandardCharsets.UTF_8),
+                    CSVFormat.DEFAULT.builder().setHeader(SkippedTerraMetricsCsvHeaders.class).build()) : null) {
+
+                // Process the executions by reading 100,000 rows, then processing the rows in parallel
+                for (CSVRecord workflowMetricRecord: workflowMetricRecords) {
+                    final int batchSize = 100000;
```

Comment:
Nit: the compiler is I'm sure smart enough to reinitialize this every time, but I'd still move it out of the loop.